main.py

```diff
@@ -16,11 +16,7 @@
     # Create logging directory
     try:
         if not os.path.exists(args.log_dir):
-            try:
-                os.makedirs(args.log_dir)
-            except OSError as exc:  # Guard against race condition
-                if exc.errno != errno.EEXIST:
-                    raise
+            os.makedirs(args.log_dir, exist_ok=True)
```

Comment:
should probably leave explanation in comment for why the try-except is still there.